rotkehlchen/chain/ethereum/oracles/uniswap.py

```diff
@@ -0,0 +1,385 @@
+                for pool in pools:
+                    if pool != ZERO_ADDRESS:
+                        found_first_link = True
+                        output.append(pool)
+                        link_asset = asset
+                        break
+                if found_first_link:
+                    # If we have an asset lets see if we can pair with to_asset
+                    assert link_asset is not None
+                    pools = self.get_pool(link_asset, to_asset)
+                    for pool in pools:
+                        if pool != ZERO_ADDRESS:
+                            found_second_link = True
+                            output.append(pool)
+                            break
+                    if found_second_link:
+                        return output
+
+        # if we reach this point it means that we need 2 more jumps
+        if not found_first_link:
+            return []
+
+        second_link_asset = None
+        for asset in self.routing_assets:
+            if asset != to_asset:
+                pools = self.get_pool(asset, to_asset)
+                for pool in pools:
+                    if pool != ZERO_ADDRESS:
+                        found_second_link = True
+                        second_link_asset = asset
+                        output.append(pool)
+                        break
+                if found_second_link:
+                    break
+
+        if not found_second_link:
+            return []
+
+        assert second_link_asset is not None
+        assert link_asset is not None
+        pools = self.get_pool(link_asset, second_link_asset)
+        for pool in pools:
+            if pool != ZERO_ADDRESS:
+                output.insert(1, pool)
+                return output
+
+        return []
+
+    def get_price(
+        self,
+        from_asset: Asset,
+        to_asset: Asset,
+        block_identifier: BlockIdentifier,
+    ) -> Price:
+        """
+        Return the price of from_asset to to_asset at the block block_identifier.
+        External oracles are used if non eth tokens are used.
+
+        Can raise:
+        - PriceQueryUnsupportedAsset
+        - RemoteError
+        """
+        log.debug(f'Searching price for {from_asset} to {to_asset} at {block_identifier!r}')
+        if not (from_asset.is_eth_token() or to_asset.is_eth_token()):
+            raise PriceQueryUnsupportedAsset(
+                f'Neither {from_asset} nor {to_asset} are ethereum tokens for the uniswap oracle',
+            )
+
+        # Use weth internally
+        if from_asset == A_ETH:
+            from_asset = A_WETH
+        if to_asset == A_ETH:
+            to_asset = A_WETH
+
+        if from_asset == to_asset:
+            return Price(ONE)
+
+        # If we are working with non eth tokens just make as if we want to go from eth/to eth
+        # and then find the price of asset <=> eth from a different oracle
+        from_asset_aux: Union[Asset, EthereumToken] = from_asset
```

Comment:
This logic here is to find a route in the case there is not a pool of `ASSET A/ ASSET B` using `ASSET A/ASSET C` and `ASSET C/ASSET B`

Reply:
I've added comments that hopefully make the code clearer and also moved parts that were repeating to aux functions. The idea is to prevent the case where the asset is an EthereumToken but as instance of `Asset` instead of `EthereumToken`